modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/issue/IssueResource.java

```diff
@@ -479,21 +483,36 @@ private UpdateIssueParams generateUpdateIssueParams( final UpdateIssueJson json
 
     private PrincipalKeys filterInvalidAssignees( final List<PrincipalKey> assignees )
     {
-        return PrincipalKeys.from( assignees.stream().filter( this::isValidAssignee ).collect( Collectors.toList() ) );
+        final ProjectPermissions projectPermissions = getCurrentProjectPermissions();
+
+        return PrincipalKeys.from( assignees.
+            stream().
+            filter( ( principalKey ) -> isValidAssignee( principalKey, projectPermissions ) ).
+            collect( Collectors.toList() ) );
     }
 
-    private boolean isValidAssignee( final PrincipalKey principalKey )
+    private ProjectPermissions getCurrentProjectPermissions()
     {
-        final PrincipalKeys membershipKeys = securityService.getAllMemberships( principalKey );
-        if ( membershipKeys.getSize() > 0 )
+        final ProjectName projectName = ProjectName.from( ContextAccessor.current().getRepositoryId() );
+        return !ProjectConstants.DEFAULT_PROJECT_NAME.equals( projectName ) ? projectService.getPermissions( projectName ) : null;
```

Comment:
It is hard for me to read this due to many negations and several exit points.
I will note other needless negations as well